hibernate-core/src/main/java/org/hibernate/boot/registry/classloading/internal/ClassLoaderServiceImpl.java

```diff
@@ -157,19 +137,96 @@ private static ClassLoader locateTCCL() {
 		}
 	}
 
+	@Override
+	public TCCLLookupBehavior getTTCLLookupBehavior() {
+		return getAggregatedClassLoader().tcclLookupBehavior;
+	}
+
+	@Override
+	public void setTCCLLookupBehavior(TCCLLookupBehavior behavior) {
+		getAggregatedClassLoader().tcclLookupBehavior = behavior;
+	}
+
 	private static class AggregatedClassLoader extends ClassLoader {
 		private final ClassLoader[] individualClassLoaders;
+		private volatile TCCLLookupBehavior tcclLookupBehavior = TCCLLookupBehavior.AFTER;
```

Comment:
just curious, why did you opt for a volatile here? Do you expect to want to change the precedence at runtime?

Would be nice to see if this could be made a `final` field.